### DISCUSSION TITLE
Community Feature: Guided User Journeys through cloud vendor kubernetes platforms

### DISCUSSION THREAD

**Discussion by kannon92** [armadaproject/armada#1804] [Ideas]
During Kubecon, there was requests about our support with managed Kubernetes Instances.  

I think we need deployment/production guides for EKS, GKE, AKE and Openshift.  

We would love community help on these as we are not focused on the cloud right now.

**Comment by Sharpz7** [comment]
This is maybe another usecase of the Armada Operator. By not relying on Terraform / Helm, we probably can support deploying on all these services directly with less config.

Link to that discussion is here: https://github.com/G-Research/armada/discussions/1796

**Reply by kannon92** [reply]
I don't think we can get away from Terraform/Helm even with an operator.  Pulsar, Redis, Postgres have helm charts and even if we use an operator, we will need a helm chart.  As this is the standard way of installing third party charts.  

I don't think we want our operator to be doing terraform installation on people's cloud vendor environments.  I think this should be separate.

**Comment by jimbobby5** [comment]
Unsure about this. Armada to date has always been quite 'bring your open Kubernetes'. We may use Terraform to stand up our own Kubernetes clusters, but the cluster configuration is deliberately quite unique to how people want to run K8s in their org (think tools / security / etc). It feels like a bit of an overreach for us to be setting up K8s itself as well

**Reply by kannon92** [reply]
My goal with this would be to mostly have a series of chart values for the kubernetes platforms so users could just apply our examples onto their existing clusters.

With our multi cluster aspect, I think we would have to maybe discuss ways of setting that up in different vendors platforms.

**Reply by jimbobby5** [reply]
I guess I don't really understand what would be different in that case between different cloud providers. The Armada layer should be configured the same on top regardless of which kind of managed Kubernetes is underneath.

**Reply by Sharpz7** [reply]
I think the point here is specifically with terraform, there is no way to configure support for all the cloud providers at once - there are completely different code settings for each provider. We should have docs for all the main ones.

**Reply by kannon92** [reply]
@Sharpz7 is correct.  I titled this "community feature" because I want this as a call to get external contributors to help us write deployment guides for platforms.  For a heathly open source project, we want to encourage contributions from other people.  

Even if GR has no interest in this, I know that the community would like us to mention how to deploy these projects to the cloud.  That usually includes guides for how to specify your helm charts and best practice for this.  Dejan made a lot of progress on EKS but I think we could follow this to the other vendors.

**Reply by jimbobby5** [reply]
Don't get me wrong, I'm not saying GR has no interest in it, I'm just trying to understand what exactly you are trying to achieve here. I think I see now
